src/view/components/TabManager/index.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { connect, MapStateToProps, MapDispatchToProps, DispatchProp } from 'react-redux';
+import { Text, Tab as NativeTab, Tabs as NativeTabs } from 'native-base';
+
+import { Tab, TabList, ElementTab, ElementTabList, StoreTabList } from '../../../store/modules/tabs/types';
+import tabActions from '../../../store/modules/tabs/actions';
+import { State } from '../../../store/types';
+
+/** TabManager OwnProps */
+interface OwnProps {}
+
+interface StoreProps {
+  activeTab: string;
+}
+
+interface DispatchProps extends DispatchProp<State> {
+  setTabs(tabs: ElementTabList, activeTab: ElementTab): void;
+  setActiveTab(activeTab: ElementTab): void;
+  clearTabs(): void;
+}
+
+interface OwnState {}
+
+/** TabManager props */
+export type Props = OwnProps & StoreProps & DispatchProps;
+
+function createSimpleComponent(tabs: ElementTabList) {
+
+  /**
+  * Page that renders different component depending on which tab the user is on
+  *
+  * Should not be instantiated directly, only extended. Unable to mark abstract due to react-redux
+  */
+  return class TabManager extends React.Component<Props, OwnState> {
+
+    /************************* Member Variables ************************/
+    private tabs: ElementTabList;
+
+    /************************* Member Functions ************************/
+
+    constructor(props: Props) {
+      super(props);
+
+      // Validation
+      if (!tabs.length) {
+        throw new Error('Cannot create TabNavigator with no tabs');
+      }
+
+      // Initialization
+      this.tabs = tabs;
+    }
+
+    private createTab(tab: ElementTab) {
+      return <NativeTab heading={tab.name} key={tab.name}>{tab.element()}</NativeTab>;
+    }
+
+    private getActiveElement() {
+      const tab = (this.props.activeTab) ? this.tabs.find(t => t.name === this.props.activeTab) : this.tabs[0];
+      if (!tab) throw new Error('Active tab is not a valid tab');
+
+      return tab.element();
+    }
+
+    /************************* React Lifecycle *************************/
```

Comment:
are these comments necessary?

Reply:
They might not be strictly necessary but I think it improves readability. You don't have to interpret which ones are react lifecycle hooks and which are redux and which belong to the class itself